src/Orleans.Transactions/Hosting/SiloBuilderExtensions.cs

```diff
@@ -8,38 +9,63 @@ namespace Orleans.Transactions
 {
     public static class SiloBuilderExtensions
     {
+        /// <summary>
+        /// Configure cluster to use an in-cluster transaction manager using defaults.
+        /// </summary>
+        public static ISiloHostBuilder UseInClusterTransactionManager(this ISiloHostBuilder builder)
+        {
+            return builder.UseInClusterTransactionManager(options => { });
```

Comment:
nit: a better way would be `builder.UseInClusterTransactionManager(default(Action<OptionsBuilder<TransactionsOptions>>));` so that it doesn't even attempt to create the options builder, as the delegate is null.
Not that I care that much, just for future work, or if you are going to address the other comment too

Reply:
btw, this is not what I meant. What I meant is passing a `null` delegate entirely, not a cached no-op delegate (no reason to cache this single use delegate and keep it forever in memory). This way `configureOptions?.Invoke(services.AddOptions<TransactionsOptions>());` will not even attempt to create an options builder, nor it would register dummy objects in the container. That whole part will be skipped and the container stay clean.

Not a big deal, but if we were keeping a dummy object, I'd prefer not caching it

---
In reply to: [154483640](https://github.com/dotnet/orleans/pull/3731#discussion_r154483640) [](ancestors = 154483640)